calculator.rb

```diff
@@ -0,0 +1,149 @@
+# Calculator program - Corinne Pingul
+
+# WELCOME----------------------------------------------------------------------
+puts """Hello there! I am a calculator, below are the operations I know how to compute: \n
+  ADDITION --> type \"add\" or \"+\"
+  SUBTRATION --> type \"sub\" or \"-\"
+  MULTIPLICATION --> type \"mult\" or \"*\"
+  DIVISION --> type \"div\" or \"/\"
+  POWERS --> type \"exp\" or \"^\"
+  """
+# QUESTIONS----------------------------------------------------------------------
+
+# Asks user what type of math they'd like to do
+print "What type of calculation would you like to do? "
+operation = gets.chomp.strip
+
+# Prints error message if user types something that's not an accepted operation
+while (operation != "add") && (operation != "+") && (operation != "sub") && (operation != "-") && (operation != "mult") && (operation != "*") && (operation != "div") && (operation != "/") && (operation != "exp") && (operation != "^")
+  print "That is not an available command. Please enter a correct operation: "
+  operation = gets.chomp.strip
+end
+
+# Gets first number from user (string)
+print "What is your first number: "
+first_num = gets.chomp
+
+# Prints error message if not a number
+while first_num != "0" && first_num.to_i == 0
```

Comment:
Figuring out how to do this validation in class was the highlight of my day. It definitely could use a clarifying comment though, for folks who weren't in the classroom with us on that glorious afternoon. :)